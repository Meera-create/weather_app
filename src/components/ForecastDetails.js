import React from "react";

function ForecastDetails(props){

    const{
        humidity,
        direction,
        windSpeed,
        temperature,
        date
    }=props;
   
    const formattedDate=new Date(date).toDateString();
    
    return(
        <div className="forecast-details" data-testid="forecast-details">
            <div className="forecast-details_humidity">
                {humidity}
            </div>
            <div className="forecast-details_direction">
                {direction}
            </div>
            <div className="forecast-details_windspeed">
                {windSpeed}
            </div>
            <div className="forecast-details_temperature">
                {temperature.max} 
                &deg;C
                
            </div>
            <div className="forecast-details_date">
                {formattedDate}
            </div>

        </div>
    )
};


export default ForecastDetails;