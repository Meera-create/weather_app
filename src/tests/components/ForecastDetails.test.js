import React from "react";
import {render} from  "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("tests forecast details",()=>{

    const Validprops3={
        temperature:{
            max:22,
            min:13
        },
        windSpeed:7,
        date:111111,
        direction:"north",
        humidity:70
    }

    it("renders the details component correctly",()=>{
        const {asFragment} = render(
            <ForecastDetails
                date={Validprops3.date}
                temperature={Validprops3.temperature}
                windSpeed={Validprops3.windSpeed}
                direction={Validprops3.direction}
                humidity={Validprops3.humidity}
            />
        )

    expect(asFragment()).toMatchSnapshot();
    })

    
    it("renders correct values for props",()=>{
        const {getByText} = render(
            <ForecastDetails
            temperature={Validprops3.temperature}
            windSpeed={Validprops3.windSpeed}
            date={Validprops3.date}
            direction={Validprops3.direction}
            humidity={Validprops3.humidity}
            
            />
        );
    //console.log(Validprops3.temperature, "HELLOOOO")
    expect(getByText("22Â°C")).toHaveAttribute("class","forecast-details_temperature")
    expect(getByText("7")).toHaveAttribute("class","forecast-details_windspeed")
    expect(getByText("Thu Jan 01 1970")).toHaveAttribute("class","forecast-details_date")
    expect(getByText("north")).toHaveAttribute("class","forecast-details_direction")
    expect(getByText("70")).toHaveAttribute("class","forecast-details_humidity")
        })

})