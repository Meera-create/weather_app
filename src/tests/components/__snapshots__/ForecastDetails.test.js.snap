// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests forecast details renders the details component correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details_humidity"
    >
      70
    </div>
    <div
      class="forecast-details_direction"
    >
      north
    </div>
    <div
      class="forecast-details_windspeed"
    >
      7
    </div>
    <div
      class="forecast-details_temperature"
    >
      22Â°C
    </div>
    <div
      class="forecast-details_date"
    >
      Thu Jan 01 1970
    </div>
  </div>
</DocumentFragment>
`;
